test/integration_filter_test.go

```diff
@@ -829,33 +832,54 @@ func TestFilterFlagWithRunAllGitFilter(t *testing.T) {
 	helpers.CleanupTerraformFolder(t, tmpDir)
 
 	testCases := []struct {
-		name          string
-		filterQuery   string
-		description   string
-		expectedUnits []string
-		excludedUnits []string
-		expectError   bool
+		name               string
+		filterQuery        string
+		description        string
+		expectedUnits      []string
+		ignoredUnits       []string
+		expectedExcluded   []string
+		filterAllowDestroy bool
+		expectError        bool
 	}{
 		{
-			name:          "git filter discovers modified, created, and removed units and excludes untouched",
-			filterQuery:   "[HEAD~1...HEAD]",
-			expectedUnits: []string{"unit-to-be-created", "unit-to-be-modified", "unit-to-be-removed"},
-			excludedUnits: []string{"unit-to-be-untouched"},
-			expectError:   false,
-			description:   "Git filter should discover units that were created, modified, or removed between commits, and exclude untouched units",
+			name:               "git filter discovers modified, created, and removed units and excludes untouched",
+			filterQuery:        "[HEAD~1...HEAD]",
+			filterAllowDestroy: false,
+			expectedUnits:      []string{"unit-to-be-created", "unit-to-be-modified"},
+			ignoredUnits:       []string{"unit-to-be-untouched"},
+			expectedExcluded:   []string{"unit-to-be-removed"},
+			expectError:        false,
+			description:        "Git filter should discover units that were created, modified, or removed between commits, and exclude untouched units. Removed unit should be excluded without --filter-allow-destroy",
+		},
+		{
+			name:               "git filter with --filter-allow-destroy includes removed unit",
```

Comment:
Possible race condition after the introduction of multiple elements in the matrix test